cmd/fetcher/main.go

```diff
@@ -48,18 +46,22 @@ func run(ctx context.Context, root string, depth int) error {
 		return err
 	}
 	client := fetchclient.New()
+	latestVersions := make(map[string]string, len(configs))
 	for _, config := range configs {
 		if config.Source.Disabled {
 			log.Printf("skipping source: %s\n", config.Filename)
 			continue
 		}
-		newVersion, err := client.Fetch(ctx, config)
-		if err != nil {
-			return err
+		newVersion := latestVersions[config.CacheKey()]
+		if newVersion == "" {
+			newVersion, err = client.Fetch(ctx, config)
+			if err != nil {
+				return err
+			}
+			latestVersions[config.CacheKey()] = newVersion
```

Comment:
We've got a lot of plugins (grpc/protoc/etc.) which share the same source.yaml configuration. This allows us to optimize lookup of new versions to only fetch the latest one once.